posthog/hogql_queries/hogql_query_runner.py

```diff
@@ -68,8 +68,17 @@ def to_actors_query(self) -> ast.SelectQuery | ast.SelectSetQuery:
     def calculate(self) -> HogQLQueryResponse:
         query = self.to_query()
         paginator = None
-        if isinstance(query, ast.SelectQuery) and not query.limit:
-            paginator = HogQLHasMorePaginator.from_limit_context(limit_context=self.limit_context)
+        if isinstance(query, ast.SelectQuery):
+            provided_limit = query.limit.value if isinstance(query.limit, ast.Constant) else None
+            provided_offset = query.offset.value if isinstance(query.offset, ast.Constant) else None
+            paginator = HogQLHasMorePaginator.from_limit_context(
+                limit_context=self.limit_context, limit=provided_limit, offset=provided_offset
+            )
+            if self.query.pagination:
```

Comment:
Here we override a SQL set LIMIT and OFFSET by a values set in pagination config, am I right? 

This should get documented somewhere, otherwise some people may get confused a bit.